packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -401,15 +445,16 @@ class SliverChildBuilderDelegate extends SliverChildDelegate {
     }
     if (child == null)
       return null;
+    final Key key = child.key != null? _SaltedValueKey(child.key) : null;
     if (addRepaintBoundaries)
-      child = RepaintBoundary.wrap(child, index);
+      child = RepaintBoundary(child: child, key: key);
     if (addSemanticIndexes) {
       final int semanticIndex = semanticIndexCallback(child, index);
       if (semanticIndex != null)
-        child = IndexedSemantics(index: semanticIndex + semanticIndexOffset, child: child);
+        child = IndexedSemantics(index: semanticIndex + semanticIndexOffset, child: child, key: key);
     }
     if (addAutomaticKeepAlives)
-      child = AutomaticKeepAlive(child: child);
+      child = AutomaticKeepAlive(child: child, key: key);
```

Comment:
we're using this key on a lot of nodes, which is a bit weird...

Reply:
Since the boolean might have different combination, I can wrote a complex if else case to determine what is the last wrap and assign key there, but I think that is kind of ugly. Do you have other suggestion?